### DISCUSSION TITLE
Quire v1.0.0-rc.0 links

### DISCUSSION THREAD

**Discussion by siobhan559** [thegetty/quire#638] [Get Help] [answered]
Hi, I'm at the V&A in London and we're currently working on a prototype of some of our old research journals (and hopefully publishing new research journals in the future) using Quire, and really loving it so far!

I just upgraded to v1.0.0-rc.0 and everything works great locally. However, I'm deploying it to an internal server for QA etc which is where I'm running into a problem. The url we want to use is https://beta.vam.ac.uk**/journals**/<contents> etc but all the links are resolving to https://beta.vam.ac.uk/<contents> for example (and the CSS is also not loaded). 

I know while using the hugo version we had a similar problem which was solved by setting relativeURLs to false in config/site.yml but there doesn't seem to be a similar setting currently. I also had a play around in the eleventy.js file with the HTML base plugin and pathPrefix settings but either my lack of understanding of eleventy or just barking up the wrong tree stopped me from getting anywhere.

Obviously the docs/upgrade is still a work in progress but I was wondering if this was a known issue?
Thanks!
Siobhan

**Comment by geealbers** [comment]
HI @siobhan559, thanks for reaching out. I'm afraid we haven't gotten far enough on testing and documenting various deploys scenarios yet, but let's see what we can figure out. To start, in your project‘s `content/_data/publication.yaml` file there is a `url` attribute at the top, is that currently set to your full deploy url?

**Reply by siobhan559** [reply]
Hey, yeah that's set to "http://beta.vam.ac.uk/journals/"

**Reply by Erin-Cecele** [reply]
Hi @siobhan559, apologies for any SPAM you may have just received from our forum. We are addressing the issue.

**Reply by geealbers** [reply]
@siobhan559 Apologies for the delay in getting back to you over the holiday break. I hope you had a happy new years! I've been doing some testing here and am seeing the same result as you. I'm going to escalate this to a bug on our issues board, and will discuss with our engineers at our team meeting this coming Monday. More to come!

**Comment by cbutcosk** [comment]
@geealbers What's the status on this issue at the moment? We've been encountering something similar when we try to deploy a Quire publication to an URL with a path in it. We had been planning on using URLs shaped like `https://publications.example.org/publication-name-slug/`, but indeed none of these options seemed to work for that: 

- `url` in `publication.yaml`, which only seemed to change the IIIF asset URLs
- `pathPrefix` in `.eleventy.js`, which in conjunction with `EleventyHtmlBasePlugin` (h/t @siobhan559 above) would emit the correctly pathed link URLs (but wouldn't emit the assets themselves from the correct URL in dev, nor in the bundled CSS itself) 
- `base` in the options passed to the Vite plugin (and the `base` option passed to the dev server), which never seemed to do much but break the asset bundler (ie, the emitted URL paths in the CSS output don't include the additional path when it would successfully build)

We also encountered a special bonus bug: `quire clean` does not (in 1.0.0-rc3 anyway) remove build products from the `.11ty-vite` directory, nor `public` so build commands can sometimes use stale assets.

**Reply by lizneely** [reply]
Hi @geealbers @Erin-Cecele !  I'm just re-checking in on @cbutcosk / @siobhan559  issues/questions and wonder if there is a timeline for resolution. We have a 11ty/Quire publication (in English and Spanish) ready to go and wonder if we need to change our publication URL plans. Thanks, Liz

**Reply by geealbers** [reply]
@lizneely Our engineers have been working through some other issues, but we can escalate this one, and I can give you a timeline. When you say 'ready to go', when exactly are you hoping to deploy?

**Reply by lizneely** [reply]
Thanks @geealbers, we are ready to deploy, but that process takes some doing surrounding getting it under the right URL. We can't start that process not knowing if we can use our preferred folder-based plan (publications.okeeffemuseum.org/exhibiting-okeeffe/, publications.okeeffemuseum.org/exhibiting-okeeffe/es/; publications.okeeffemuseum.org/halvorson/; publications.okeeffemuseum.org/halvorson/es/, etc.) We are stuck unless we give all of those publications unique subdomain URLs, which isn't ideal. Do you know when this will be addressed and is there an estimate for how long it will take to resolve? @cbutcosk

**Reply by geealbers** [reply]
@lizneely and @cbutcosk, our engineers are off tomorrow, but we have our team meeting with them on Monday at 11:30. We'll get an ETA on the fix then and let you know. Sorry for the trouble. And we're excited to see your publication once we can get you past this bug!

**Reply by lizneely** [reply]
Thanks @geealbers -- let us know! We can't wait to share! Design for Context has connected this into our Linked Art dataset so we'll have 2-way discoverability of collections references and a data index. It's pretty nifty!

**Reply by cbutcosk** [reply]
@geealbers Thanks for escalating this issue. We ended up working around it for the moment via `HtmlBasePlugin` with some additional bundler config and some template changes to correct misfiring URL math and filters in the quire base templates. 

Let us know if you'd like a pull request, but also keep us in the loop as your team comes to a final solve. It's important to us to keep this publication in-sync with quire `main`'s deployment strategies and config (especially as far as injecting the path from the pub config into the 11ty config goes), at least until you all are 1.0-GM.

For others encountering this issue (@siobhan559) the core of the fix was:
- In `.eleventy.js`, add `EleventyHtmlBasePlugin` and init with no options:
```js
  /**
   * @see https://www.11ty.dev/docs/plugins/html-base/
   */
  eleventyConfig.addPlugin(EleventyHtmlBasePlugin)
```
- In `.eleventy.js`, set `prefixPath` to the prefix path of the (eg, `ex.org/a-great-publication/` would have a `prefixPath` of `/a-great-publication/`
```js
    /**
     * @see {@link https://www.11ty.dev/docs/config/#deploy-to-a-subdirectory-with-a-path-prefix}
     */
    pathPrefix: '/a-great-publication/',
```
- In `.eleventy.js`, in the vite config options, add a `resolver.alias` key that aliases the prefixPath to root (`/`):
```js
eleventyConfig.addPlugin(EleventyVitePlugin, {
    tempFolderName: '.11ty-vite',
    viteOptions: {
      publicDir: process.env.ELEVENTY_ENV === 'production'
        ? publicDir
        : false,
      /**
       * @see https://vitejs.dev/config/#build-options
       */
      root: outputDir,
      base: '/a-great-publication/',
      resolve: {
        alias: [{find: '/a-great-publication/', replacement: "/"}]
      },
      build: {
...
```
- In `publication.yaml` we set the url to the full url of the publication with a trailing slash (eg, `https://ex.org/a-great-publication/`).

WIth all that, `quire preview` should tell you the right path to hit for dev previews when you start, and visiting localhost:8080/a-great-publication/ should show the whole publication with the correct CSS loaded. The main template issues I hit were in the TOC items, search init (the payload path and the item pathing serialized in the index), and citation URLs in the menu footer. Cheers!

ETA: Corrected github handle
ETA2: Added `base` to vite config

**Reply by hbalenda** [reply]
@cbutcosk Thank you for investigating - we would love to collaborate on this if you don't mind opening a PR with the work you've already done. Additionally, we will want to set the `pathPrefix` from the `url` value in `publication.yaml` so that users don't have to edit the `.eleventy.js` to use the path prefix, and if you open the pull request I can also address the remaining issues you outlined (TOC links, search, citations) there before we merge this into `main`. Thank you again!

**Reply by cbutcosk** [reply]
@hbalenda Submitted as a PR, let me know if there's anything I can do to help that along. Cheers!

**Comment by cbutcosk** [comment]
@hbalenda In updating our pub to quire-RC7 I noticed there's still one missing fix here in quire `main`--https://github.com/thegetty/quire/blob/1c8aaea1faf701ed6236fcd4bea73785f0db7a04/packages/11ty/_includes/search.liquid#L12 should reflect the prefixed path:
```liquid
<div
  aria-expanded="false"
  class="quire-search"
  data-search-index="{{ '/search-index.json' | htmlBaseUrl }}"
  id="js-search"
>
```

**Reply by hbalenda** [reply]
Ah, thanks @cbutcosk for catching this, will submit a fix soon.

**Reply by chrisdaaz** [reply]
Is quire-RC7 out? I'm finishing up a project with RC3 but I will update and rebuild with RC-7 if this is included. We're hoping to publish to a subpath rather than subdomain.

**Reply by cbutcosk** [reply]
They mostly seem to do releases on npm at the moment: https://www.npmjs.com/package/@thegetty/quire-11ty

**Reply by chrisdaaz** [reply]
Ah, thanks! I was looking at `@thegetty/quire-cli` (rc.6) on npm, not realizing it installs `@thegetty/quire-11ty` (rc.7).

**Comment by Erin-Cecele** [comment, marked as answer]
This was fixed with PR #664 (which addresses Issue #642). Thank you, @cbutcosk, for your collaboration with the Quire team to get this fixed!
